scripts/o3de/o3de/utils.py

```diff
@@ -109,6 +112,22 @@ def validate_identifier(identifier: str) -> bool:
     return True
 
 
+def validate_version_specifier(version_specifier:str) -> bool:
+    try:
+        get_object_name_and_version_specifier(version_specifier)
+    except Exception as e:
+        return False
```

Comment:
This is dangerous as it swallows all exceptions, even ones we didn't expect that could hide bugs in the code